src/core/internal/abort.d

```diff
@@ -0,0 +1,44 @@
+module core.internal.abort;
+
+/*
+ * Use instead of assert(0, msg), since this does not print a message for -release compiled
+ * code, and druntime is -release compiled.
+ */
+void abort(string msg, string filename = __FILE__, size_t line = __LINE__) @nogc nothrow @safe
+{
+    // use available OS system calls to print the message to stderr
+    version(Posix)
+    {
+        import core.sys.posix.unistd: write;
+        static void writeStr(const(char)[][] m...) @nogc nothrow @trusted
+        {
+            foreach(s; m)
+                cast(void)write(2, s.ptr, s.length);
```

Comment:
What it does is tell the compiler "yes, I know this returns a value and I'm not using it".

For pure functions with no mutable reference parameters, this makes a lot of sense (why call it otherwise?)